pkg/operator/starter.go

```diff
@@ -255,6 +261,18 @@ func RunOperator(ctx context.Context, controllerConfig *controllercmd.Controller
 		controllerConfig.Server.Handler.NonGoRestfulMux.Handle("/debug/controllers/resourcesync", debugHandler)
 	}
 
+	// Ensure removal of the daemonset that was previously used to manage apiserver pods.
+	dsClient := kubeClient.AppsV1().DaemonSets(operatorclient.TargetNamespace)
+	// TODO(marun) Should deletion block?
+	go wait.PollImmediateUntil(time.Minute, func() (bool, error) {
```

Comment:
does this mean we unconditionally remove the daemonset before we know that the deployment is up already? We might want to check that at least one pod of the deployment is healthy.

Reply:
Done.